libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/SpanContext.java

```diff
@@ -31,4 +31,19 @@ public SpanContext(Span span) {
     Span getSpan() {
         return span;
     }
+
+    /**
+     * Sets the error for the current span behind this context
+     * @param cause error
+     */
+    public void setError(final Exception cause) {
+        span.setError(cause);
+    }
+
+    /**
+     * Ends current span
+     */
+    public void endSpan() {
```

Comment:
In case we need a mutable behaviour on the SpanContext then better we should start returning the Span itself from the tracer::getCurrentSpan. I can may be take this up in a separate PR.

Reply:
Yeah, let see if we need that, looks like one-off for now